### PR TITLE
Update RPO with a comment on security given domain separation

### PR DESCRIPTION
**Pull request by Al-Kindi-0**
## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: src/hash/rescue/rpo/mod.rs [hunk 1]
```diff
@@ -52,6 +53,17 @@ mod tests;
 /// to deserialize them into field elements and then hash them using
 /// [hash_elements()](Rpo256::hash_elements) function rather then hashing the serialized bytes
 /// using [hash()](Rpo256::hash) function.
+///
+/// ## Domain separation
```
Comment (bobbinth) [inline src/hash/rescue/rpo/mod.rs:57]: Should we make similar comments for the [Rpx256](https://github.com/0xPolygonMiden/crypto/blob/50b5bf1d875e995ae129eec54d55b752a8586c37/src/hash/rescue/rpx/mod.rs#L59) struct?

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left a couple of small comments inline. Once these are addressed, we can merge.

File: CHANGELOG.md
[the remark this answers is not in the document]
Comment (Al-Kindi-0) [reply CHANGELOG.md:1]: It happened now a couple of times in different PRs so I am not sure why.
Fixed it back.

File: src/hash/rescue/rpo/mod.rs (hunk 1 above)
Comment (Al-Kindi-0) [reply src/hash/rescue/rpo/mod.rs:57]: Add an adapted comment to RPX

**Review by bobbinth — APPROVED** [review]
All looks good! Thank you!

**Pull request merged**